core/trino-main/src/main/java/io/trino/server/protocol/spooling/QueryDataEncodingConfig.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.server.protocol.spooling;
+
+import com.google.common.collect.ImmutableSet;
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+
+import java.util.Set;
+
+public class QueryDataEncodingConfig
+{
+    private Set<String> disabledEncodings = ImmutableSet.of();
+
+    public Set<String> getDisabledEncodings()
+    {
+        return disabledEncodings;
+    }
+
+    @Config("protocol.spooling.disabled-encodings")
+    @ConfigDescription("Disabled spooled protocol encodings")
+    public QueryDataEncodingConfig setDisabledEncodings(Set<String> disabledEncodings)
```

Comment:
You may consider going with multiple config options 
`protocol.spooling.encoding.<name>.enabled=...`

Reply:
what does it buy us?

Reply:
Actually, it makes sense.